app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,14 @@
+class SessionsController < ApplicationController
+  def create
+    user = User.from_omniauth(request.env['omniauth.auth'])
+
+    if user.persisted?
+      flash[:success] = "You're logged in #{user.first_name}"
+      session[:user_id] = user.id
+      redirect_to new_search_path
+    else
+      flash[:error] = "Something happened, please try again"
+      redirect_to root_path
```

Comment:
These flash messages are a nice touch! We should add these kinds of details in our user stories!